New

DouglasBurns opened the pull request:
Pull request if you want...

DouglasBurns committed:
Store B-quark information
Store single lepton information
Store neutrino information (GEN levels)

DouglasBurns committed:
Resetting back

DouglasBurns committed:
hope it works

DouglasBurns committed:
fix stash

DouglasBurns committed:
Likelihood analyser:
Store Neutrino ChiSq Sol
Store Mass Top, Mass W for (in)correct permutations of B Jets
Store CSV Jets/BJets

EmyrClement committed:
Read in file containing likelihood inputs.

EmyrClement committed:
Start adding likelihood ttbar reco module.

EmyrClement committed:
Add analyser for likelihood reco method.

EmyrClement committed:
Few changes to Likelihood reco module.

EmyrClement committed:
Add likelihood analyser to ttbar plus x analyser.

EmyrClement committed:
Debug reading in file.

EmyrClement committed:
Add example input file for likelihood ttbar reco.

DouglasBurns committed:
Source Analyser changed

DouglasBurns committed:
Store data to look at Top Variable Plots
Store data to look at Likelihood Plots (Separately for Mass/CSV/NuChi2)
Store type of Reconstruciton Solution (Correct/Incorrect/Not Reconstrcutible/Not Semileptonic)

EmyrClement commented on src/Analysers/LikelihoodInputAnalyser.cpp:
```diff
@@ -15,34 +15,198 @@ void LikelihoodInputAnalyser::analyse(const EventPtr event) {
 	weight_ = event->weight();
 	treeMan_->setCurrentFolder(histogramFolder_);
+	unsigned int parton = 0;
+	double CSV;
```
Initialise to something!

EmyrClement commented on src/Analysers/LikelihoodInputAnalyser.cpp:
```diff
@@ -15,34 +15,198 @@ void LikelihoodInputAnalyser::analyse(const EventPtr event) {
+			// 	treeMan_->Fill("JetPt", AllJet->pt());
+			// 	treeMan_->Fill("JetEta", AllJet->eta());
+			// }
+
```
Remove this commented bit if not actually used.

EmyrClement commented on src/Analysers/LikelihoodInputAnalyser.cpp:
```diff
@@ -15,34 +15,198 @@ void LikelihoodInputAnalyser::analyse(const EventPtr event) {
+
+			if ( abs(CSV) >= 11 ){
+				cout << "CSV : " << CSV << " Run Number : " << event->runnumber() << " Event Number : " << event->eventnumber() << " Lumi Number : " << event->lumiblock() << endl;
+				// event->inspect(); //For more info on events failing
```
Comment out or remove cout statement.

EmyrClement commented on src/Analysers/LikelihoodInputAnalyser.cpp:
```diff
@@ -15,34 +15,198 @@ void LikelihoodInputAnalyser::analyse(const EventPtr event) {
+				// cout << "LightJet CSV : " << CSV << endl;
+				if ( CSV == -10 ){
+					treeMan_->Fill("LightJetPt", AllJet->getFourVector().Pt());
+					treeMan_->Fill("LightJetEta", AllJet->getFourVector().Eta());
```
Remove these plots for now?  A bit misleading if you fill a branch called "LightJetPt" and only jets that have weird CSV are considered.

EmyrClement commented on src/Analysers/LikelihoodInputAnalyser.cpp:
```diff
@@ -15,34 +15,198 @@ void LikelihoodInputAnalyser::analyse(const EventPtr event) {
+				if ( CSV == -10 ){
+					treeMan_->Fill("BJetPt", AllJet->getFourVector().Pt());
+					treeMan_->Fill("BJetEta", AllJet->getFourVector().Eta());
+				}
```
Same as above.

DouglasBurns committed:
Updated LikelihoodInputAnalyser.cpp with changes

EmyrClement commented on src/Analysers/LikelihoodRecoAnalyser.cpp:
```diff
@@ -0,0 +1,180 @@
+		if ( ttbarpartonsfound != 4 ){
+			TypeofSolution = 3;
+			treeMan_->Fill("TypeofSolution", TypeofSolution);
+		}
```
It looks like you might be filling the TypeOfSolution branch a number of times per event?  e.g. if the event passes offline selection but isn't semileptonic (so type of solution = 4), you probably won't find all the ttbar_decay_partons, so type of solution is also = 3.  I'm not sure what happens if you call Fill on the same branch twice in one event...

I think what you need to do is :

if ( ttbarpartonsfound != 4 && selectionCriteria > 0 ) ?

EmyrClement commented on src/Analysers/LikelihoodRecoAnalyser.cpp:
```diff
@@ -0,0 +1,180 @@
+			}
+			treeMan_->Fill("EventReconstruction", EventCorrect);
+		}
+
```
I would suggest only doing treeMan_->Fill("TypeofSolution", TypeofSolution) once per event, to make sure you only fill it once.  Set TypeofSolution as you work through all the different categories, and then fill the branch once.

EmyrClement commented on src/Analysers/TTbar_plus_X_analyser.cpp:
```diff
@@ -95,6 +96,8 @@ void TTbar_plus_X_analyser::ePlusJetsSignalAnalysis(const EventPtr event) {
 
+		likelihoodRecoAnalyserEPlusJetsRefSelection_->analyse(event, jets, bJets, signalLepton, MET_main);
+
 		// ref_selection_binned_HT_analyser_electron_->setScale(bjetWeight * efficiencyCorrection);
```
Just noticed your new reco analyser is not run in the muon channel.  You should add it in under TTbar_plus_X_analyser::muPlusJetsSignalAnalysis

EmyrClement commented on src/Readers/TTGenReader.cpp:
```diff
@@ -160,6 +160,7 @@ void TTGenReader::readTTGenInfo() {
 
     ttGenInfo_ = TTGenInfoPointer( new TTGenInfo( quark, quarkBar, leptonicB, hadronicB, singlelepton, singleneutrino, qGenJetIndex, qBarGenJetIndex, leptonicBGenJetIndex, hadronicBGenJetIndex ) );
+
 }
```
Remove this addition.

DouglasBurns committed:
Extended Changes to:
LikelihoodInputAnalyser.cpp
LikelihoodRecoAnalyser.cpp
TTbar_plus_X_analyser.cpp
TTGenReader.cpp

Changed way type of solution was stored
Included analysis in Mu channel
Generally cleaned up code
